src/main/scala/elevate/core/strategies/predicate.scala

```diff
@@ -7,27 +7,40 @@ import scala.language.implicitConversions
 
 object predicate {
 
-  implicit def rewriteResultToBoolean[P](r: RewriteResult[P]): Boolean = r match {
-    case Failure(_) => false
-    case Success(_) => true
-  }
+  implicit def rewriteResultToBoolean[P](r: RewriteResult[P]): Boolean =
+    r match {
+      case Failure(_) => false
+      case Success(_) => true
+    }
 
   case class not[P](s: Strategy[P]) extends Strategy[P] {
     def apply(e: P): RewriteResult[P] = s(e) match {
       case Success(_) => Failure(not(s))
       case Failure(_) => Success(e)
     }
-    override def toString = s"not($s)"
+    override def toString: String = s"not($s)"
   }
 
+  // only fails if the else case fails
+  // id is required to not trigger the else case if the then case fails
+  def ifThenElse[P](p: Strategy[P],
+                    t: Strategy[P],
+                    e: Strategy[P]): Strategy[P] =
+    (p `;` (t <+ basic.id[P]())) <+ e
```

Comment:
It would seem more consistent to neither fail for `t` nor for `e`? Only failing for `e` seems very arbitrary and weird.